src/mods/storytimer.cpp

```diff
@@ -0,0 +1,368 @@
+            s_display_segment_timer = false;
+            break;
+    }
+
+    // if the segment timer is enabled in any capacity, show all 10 split times + iw times after the
+    // tape is broken on the last stage
+    if (TimerOptions(pref::get(pref::U8Pref::SegmentTimerOptions)) != TimerOptions::DontShow &&
+        is_run_complete) {
+        // haven't attempted the mkb::sprintf suggestion yet
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y, IW_TIME_TEXT_OFFSET,
+                              "W1:", s_timer_group[0].split, s_timer_group[0].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 1, IW_TIME_TEXT_OFFSET,
+                              "W2:", s_timer_group[1].split, s_timer_group[1].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 2, IW_TIME_TEXT_OFFSET,
+                              "W3:", s_timer_group[2].split, s_timer_group[2].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 3, IW_TIME_TEXT_OFFSET,
+                              "W4:", s_timer_group[3].split, s_timer_group[3].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 4, IW_TIME_TEXT_OFFSET,
+                              "W5:", s_timer_group[4].split, s_timer_group[4].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 5, IW_TIME_TEXT_OFFSET,
+                              "W6:", s_timer_group[5].split, s_timer_group[5].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 6, IW_TIME_TEXT_OFFSET,
+                              "W7:", s_timer_group[6].split, s_timer_group[6].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 7, IW_TIME_TEXT_OFFSET,
+                              "W8:", s_timer_group[7].split, s_timer_group[7].segment, true, false,
+                              draw::WHITE);
+        timerdisp::draw_timer(IW_TIME_LOCATION_X, segment_timer_location_y + 8, IW_TIME_TEXT_OFFSET,
+                              "W9:", s_timer_group[8].split, s_timer_group[8].segment, true, false,
+                              draw::WHITE);
+        // use segment timer spacing for w10 since "W10" is 3 characters long, not 2
+        timerdisp::draw_timer(SEGMENT_TIMER_LOCATION_X, segment_timer_location_y + 9,
+                              SEGMENT_TIMER_TEXT_OFFSET, "W10:", s_timer_group[9].split,
+                              s_timer_group[9].segment, true, false, draw::WHITE);
+    }
```

Comment:
Should this be it's own bool preference? Or just the EndOfRun case?

Reply:
The way it works now is that if the segment timer is on at all (so either showing at all times, between worlds, or at the end of the run), when the run is finished, all 10 split and segment times are shown. I don't think this should get its own bool pref since this information is shown after the run, and so people who want to run pace blind aren't affected by it